Modules/Camouflague.cs

```diff
@@ -53,7 +53,15 @@ public static void CheckCamouflage()
 
             if (oldIsCamouflage != IsCamouflage)
             {
-                Main.AllPlayerControls.Do(pc => Camouflage.RpcSetSkin(pc));
+                foreach (var pc in Main.AllPlayerControls)
+                {
+                    Camouflage.RpcSetSkin(pc);
+
+                    if (!IsCamouflage && !pc.IsAlive())
+                    {
+                        pc.RpcSetPet("");
+                    }
```

Comment:
Please add a comment that explains the purpose of this code in case this is no longer necessary due to future updates on the AU side. You can write in English.